src/main/java/io/github/nucleuspowered/nucleus/modules/kit/commands/firstjoin/FirstKitRedeemCommand.java

```diff
@@ -6,41 +6,64 @@
 
 import io.github.nucleuspowered.nucleus.Util;
 import io.github.nucleuspowered.nucleus.dataservices.KitService;
-import io.github.nucleuspowered.nucleus.internal.annotations.NoCooldown;
-import io.github.nucleuspowered.nucleus.internal.annotations.NoCost;
-import io.github.nucleuspowered.nucleus.internal.annotations.NoWarmup;
-import io.github.nucleuspowered.nucleus.internal.annotations.Permissions;
-import io.github.nucleuspowered.nucleus.internal.annotations.RegisterCommand;
+import io.github.nucleuspowered.nucleus.internal.annotations.*;
 import io.github.nucleuspowered.nucleus.internal.command.AbstractCommand;
+import io.github.nucleuspowered.nucleus.internal.command.ReturnMessageException;
 import io.github.nucleuspowered.nucleus.modules.kit.config.KitConfigAdapter;
 import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
 import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.args.CommandElement;
+import org.spongepowered.api.command.args.GenericArguments;
 import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.text.Text;
 import org.spongepowered.api.util.Tristate;
 
 import javax.inject.Inject;
+import java.util.Optional;
 
 @SuppressWarnings("ALL")
 @Permissions(prefix = "firstjoinkit")
 @NoCost
 @NoCooldown
 @NoWarmup
 @RegisterCommand(value = "redeem", subcommandOf = FirstKitCommand.class)
-public class FirstKitRedeemCommand extends AbstractCommand<Player> {
-
-    @Inject
-    private KitService gds;
+public class FirstKitRedeemCommand extends AbstractCommand<CommandSource> {
+    private final String playerKey = "subject";
 
+    @Inject private KitService gds;
     @Inject private KitConfigAdapter kca;
 
     @Override
-    public CommandResult executeCommand(Player src, CommandContext args) throws Exception {
+    public CommandElement[] getArguments() {
+        return new CommandElement[] {
+                GenericArguments.onlyOne(GenericArguments.optionalWeak(GenericArguments.player(Text.of(playerKey))))
+        };
+    }
+
+    @Override
+    public CommandResult executeCommand(CommandSource src, CommandContext args) throws Exception {
+        Optional<Player> player = args.<Player>getOne(playerKey);
```

Comment:
Done! Though I did find that now the kit can't be given to a Player via console when testing.